projects/gloo/pkg/bootstrap/clients/secret.go

```
@@ -0,0 +1,295 @@
+			}
+		}
+		c, err := f.NewResourceClient(ctx, params)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to initialize secret resource client from factory of type %T", f)
+		}
+
+		multiClient.clientList = append(multiClient.clientList, c)
+	}
+	return multiClient, nil
+}
+
+type (
+	kubeSecretClientSettings struct {
+	}
+	directorySecretClientSettings struct {
+		rootDir string
+	}
+)
+
+// we should not need a RWMutex here because we are only ever writing to the map during
+// its instantion in NewResourceClient
+type MultiResourceClient struct {
+	clientList []clients.ResourceClient // do not use clients.ResourceClients here as that is not for this purpose
+}
+
+// Clients provides access to the sub-clients used by this multi-client. This is primarily
+// exposed for testing; care should be taken when mutating this client at runtime
+// as the Watch method may have spawned goroutines with clients that will not
+// be updated by the mutation.
+func (m *MultiResourceClient) Clients() []clients.ResourceClient {
+	return m.clientList
+}
+
+func (m *MultiResourceClient) Kind() string {
+	if len(m.clientList) == 0 {
+		return ""
+	}
+
+	// Any of the clients should be able to handle this identically
+	return m.clientList[0].Kind()
+}
+
+func (m *MultiResourceClient) NewResource() resources.Resource {
+	if len(m.clientList) == 0 {
+		return nil
+	}
+
+	// Any of the clients should be able to handle this identically
+	return m.clientList[0].NewResource()
+}
+
+// Deprecated: implemented only by the kubernetes resource client. Will be removed from the interface.
+func (m *MultiResourceClient) Register() error {
+	for _, v := range m.clientList {
+		switch concreteClient := v.(type) {
+		case *vault.ResourceClient:
+			continue
+		case *file.ResourceClient:
+			continue
+		default:
+			return concreteClient.Register()
+		}
+	}
+	return errors.New("Register method only implemented on Kubernetes resource client")
```

Do we need to return an error here? Not sure from where if anywhere this interface function is called